sdks/python/apache_beam/transforms/core.py

```diff
@@ -165,13 +288,29 @@ def default_label(self):
     return self.__class__.__name__
 
   def process(self, element, *args, **kwargs):
-    """Called for each element of a pipeline. The default arguments are needed
-    for the DoFnRunner to be able to pass the parameters correctly.
+    """Method to use for processing elements.
+
+    This is invoked by ``DoFnRunner`` for each element of a input
+    ``PCollection``.
+
+    If specified, following default arguments are used by the ``DoFnRunner`` to
+    be able to pass the parameters correctly.
+
+    ``DoFn.ElementParam``: element to be processed.
+    ``DoFn.SideInputParam``: a side input that may be used when processing.
+    ``DoFn.TimestampParam``: timestamp of the input element.
+    ``DoFn.WindowParam``: ``Window`` the input element belongs to.
+    An object of type ``RestrictionProvider``: having a parameter of a type that
```

Comment:
An ``iobase.RestrictionProvider`` instance: a restriction tracker will be provided here to allow treatment as a Splittable `DoFn``.

Reply:
Done.